docs/secret_management.md

```diff
@@ -0,0 +1,172 @@
+# Secrets Management for the API and APP keys
+
+The Datadog-Operator is able to retrieves the datadog credentials securely thanks to 3 different methods.
+
+## 1. Plain credentials in DatadogAgent resource
```

Comment:
```suggestion
## 1. Plain credentials in Datadog Agent resource
```

Reply:
`DatadogAgent` should be in one word. it is a Kubernetes resource name.

@apigirl is it more clear if we put:
```suggestion
## 1. Plain credentials in `DatadogAgent` kubernetes resource
```

Reply:
IMO either "DatadogAgent resource" or "DatadogAgent custom resource" should be fine. Kubernetes is a bit redundant since we are in the context of Kubernetes already